src/application/MenuViewExample.cpp

```diff
@@ -0,0 +1,120 @@
+
+    _header->getTextComponent().setDisplayString("Example Header")
+        .setFontColour(COLOUR_WHITE).setFont(FREE_SANS_12PT7B);
+    _header->setWidth(HEADER_WIDTH).setHeight(HEADER_HEIGHT).setBackgroundColour(COLOUR_BLACK);
+    
+    _menuButton1->getTextComponent().setDisplayString("Button 1").setFontSize(1,1)
+        .setFontColour(COLOUR_BLACK).setFont(FREE_SANS_BOLD_9PT7B);
+    _menuButton1->setFontFocusColour(COLOUR_WHITE).setFocusColour(COLOUR_BLACK)
+        .setWidth(MENU_BUTTON_WIDTH).setHeight(MENU_BUTTON_HEIGHT)
+        .setBackgroundColour(COLOUR_LIGHTGREY).addBorder(COLOUR_BLACK);
+
+    _menuButton2->getTextComponent().setDisplayString("Button 2").setFontSize(1,1)
+        .setFontColour(COLOUR_BLACK).setFont(FREE_SANS_BOLD_9PT7B);
+    _menuButton2->setFontFocusColour(COLOUR_WHITE).setFocusColour(COLOUR_BLACK)
+        .setWidth(MENU_BUTTON_WIDTH).setHeight(MENU_BUTTON_HEIGHT)
+        .setBackgroundColour(COLOUR_LIGHTGREY).addBorder(COLOUR_BLACK);
+
+    _menuButton3->getTextComponent().setDisplayString("Button 3").setFontSize(1,1)
+        .setFontColour(COLOUR_BLACK).setFont(FREE_SANS_BOLD_9PT7B);
+    _menuButton3->setFontFocusColour(COLOUR_WHITE).setFocusColour(COLOUR_BLACK)
+        .setWidth(MENU_BUTTON_WIDTH).setHeight(MENU_BUTTON_HEIGHT)
+        .setBackgroundColour(COLOUR_LIGHTGREY).addBorder(COLOUR_BLACK);
+
+    _navButton1->getTextComponent().setDisplayString("Back").setFontSize(1,1)
+        .setFontColour(COLOUR_WHITE).setFont(FREE_SANS_BOLD_9PT7B);
+    _navButton1->setFontFocusColour(COLOUR_WHITE).setFocusColour(COLOUR_PURPLE)
+        .setWidth(NAV_BUTTON_WIDTH).setHeight(NAV_BUTTON_HEIGHT)
+        .setBackgroundColour(COLOUR_RED).addBorder(COLOUR_BLACK);
+
+    _navButton2->getTextComponent().setDisplayString("Select").setFontSize(1,1)
+        .setFontColour(COLOUR_WHITE).setFont(FREE_SANS_BOLD_9PT7B);
+    _navButton2->setFontFocusColour(COLOUR_WHITE).setFocusColour(COLOUR_DARKCYAN)
+        .setWidth(NAV_BUTTON_WIDTH).setHeight(NAV_BUTTON_HEIGHT)
+        .setBackgroundColour(COLOUR_GREEN).addBorder(COLOUR_BLACK);
+
+    // padding
+    _window->setOrientation(Container::Column).setWidth(WINDOW_WIDTH).setHeight(WINDOW_HEIGHT)
+        .setPosition(WINDOW_TOP_LEFT).setBackgroundColour(COLOUR_BLACK).addBorder(COLOUR_RED);
+    _menuContainer->setOrientation(Container::Column).setWidth(MENU_WIDTH).setHeight(MENU_HEIGHT)
+        .setBackgroundColour(COLOUR_DARKGREY).addBorder(COLOUR_MAGENTA);
+    _navButtonContainer->setOrientation(Container::Row).setWidth(NAV_WIDTH).setHeight(NAV_HEIGHT)
+        .setBackgroundColour(COLOUR_NAVY).addBorder(COLOUR_BLACK);
+    
+    // order matters for alignment: align top-level container first, then down
+    _window->align();
+    _menuContainer->align();
+    _navButtonContainer->align();
+
+    // call to principal container draws all elements
+    _window->draw();
+}
+
+void MenuViewExample::run() {
```

Comment:
What is happening here exactly? Is this something to do with interactivity? And why is there a ```delay(1000)``` on line 115?

Reply:
This is just lazy demo code that walks through all the selectable elements of the example menu, focusing and unfocusing each one.  The delay is there to leave some time between focus / unfocus so you can see what it looks like on the display.